src/Data/String/LineWrapping.purs

```diff
@@ -0,0 +1,113 @@
+{-
+Copyright 2016 SlamData, Inc.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-}
+
+module Data.String.LineWrapping
+  ( MeasuredWord
+  , WrappedLine
+  , Conf
+  , measuredWords
+  , unwrapMeasuredWord
+  , unwrapWrappedLine
+  , wrappedLines
+  , printWrappedLine
+  , printMeasuredWord
+  , splitByNewlineOrSpace
+  ) where
+
+import Prelude
+
+import Control.Monad.Rec.Class (Step(Loop, Done), tailRec)
+import Data.Array as Array
+import Data.String as String
+import Data.Foldable (sum)
+import Data.Int as Int
+import Data.Maybe (Maybe(Just, Nothing))
+import Data.Traversable (traverse)
+
+type Conf = { maxWidth ∷ Number, spaceWidth ∷ Number }
+
+type MeasuredWordR = { string ∷ String, width ∷ Number }
+
+newtype MeasuredWord = MeasuredWord MeasuredWordR
+
+newtype WrappedLine = WrappedLine (Array MeasuredWord)
+
+derive instance eqMeasuredWord ∷ Eq MeasuredWord
+
+derive newtype instance eqWrappedLine ∷ Eq WrappedLine
+
+unwrapMeasuredWord ∷ MeasuredWord → MeasuredWordR
+unwrapMeasuredWord (MeasuredWord x) = x
+
+unwrapWrappedLine ∷ WrappedLine → Array MeasuredWord
+unwrapWrappedLine (WrappedLine x) = x
```

Comment:
`un` - we started switching away from `run` a while back (before deleting entirely in favour of `Newtype` when appropriate). We only use `run` for things like transformers, where it kinda makes a bit more sense, for data `un` seems more sensible since there's not really a concept of anything actually happening when you unwrap.